plugins/assimp/module/vtkF3DAssimpImporter.cxx

```diff
@@ -257,15 +263,23 @@ class vtkF3DAssimpImporter::vtkInternals
     }
     else
     {
-      vtkNew<vtkImageData> img;
-      img->SetDimensions(aTexture->mWidth, aTexture->mHeight, 1);
-      img->AllocateScalars(VTK_UNSIGNED_CHAR, 4);
+      // Sometimes Assimp returns corrupted textures (encountered with 3MF)
+      // Let's validate it before trying to read it
+      std::regex validRegexp("[rgba]{4}[0-9]{4}");
```

Comment:
id expect this line to be covered ?

Reply:
I still dont get the comment if you never encountered the issue ?

Reply:
I have seen some 3mf files with corrupted memory resulting in a non-zero height value (meaning raw texture) but with trash in the image format. At least the regexp ensures the specified format is what we are expecting before trying to read the memory (it's crashing in f3d otherwise)

Reply:
Cant we use such file for coverage at least ?

Reply:
No, because it crashes in assimp when the scene is destructed.

Reply:
For file `3mf-samples/examples/material/multiprop-metallic.3mf` for example, here's the values:
```
mHeight:32589
mWidth:1858039584
achFormatHint: onM
```
And the relevant assimp documentation:
```
     * If mHeight != 0 this member is show how data is packed. Hint will consist of
     * two parts: channel order and channel bitness (count of the bits for every
     * color channel). For simple parsing by the viewer it's better to not omit
     * absent color channel and just use 0 for bitness. For example:
     * 1. Image contain RGBA and 8 bit per channel, achFormatHint == "rgba8888";
     * 2. Image contain ARGB and 8 bit per channel, achFormatHint == "argb8888";
     * 3. Image contain RGB and 5 bit for R and B channels and 6 bit for G channel, achFormatHint == "rgba5650";
     * 4. One color image with B channel and 1 bit for it, achFormatHint == "rgba0010";
```

If I do not check `achFormatHint`, there's a crash in f3d because `mWidth` is too large.

Reply:
And the crash backtrace at exit:

```
#0  __pthread_kill_implementation (threadid=<optimized out>, 
    signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff76ac8a3 in __pthread_kill_internal (signo=6, 
    threadid=<optimized out>) at pthread_kill.c:78
#2  0x00007ffff765c668 in __GI_raise (sig=sig@entry=6)
    at ../sysdeps/posix/raise.c:26
#3  0x00007ffff76444b8 in __GI_abort () at abort.c:79
#4  0x00007ffff7645390 in __libc_message (
    fmt=fmt@entry=0x7ffff77bc55d "%s\n")
    at ../sysdeps/posix/libc_fatal.c:150
#5  0x00007ffff76b67b7 in malloc_printerr (
    str=str@entry=0x7ffff77ba02b "free(): invalid pointer")
    at malloc.c:5765
#6  0x00007ffff76b8a74 in _int_free (av=<optimized out>, 
    p=<optimized out>, have_lock=have_lock@entry=0) at malloc.c:4500
#7  0x00007ffff76bb353 in __GI___libc_free (mem=<optimized out>)
    at malloc.c:3391
#8  0x00007ffff78aeaba in operator delete (ptr=<optimized out>)
    at /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/del_op.cc:49
#9  0x00007ffff78aeaea in operator delete[] (ptr=<optimized out>)
    at /usr/src/debug/gcc/gcc/libstdc++-v3/libsupc++/del_opv.cc:35
#10 0x00007fffed78ff09 in aiTexture::~aiTexture (this=<optimized out>, 
    this=<optimized out>)
    at /usr/src/debug/assimp/assimp-5.3.1/include/assimp/texture.h:212
#11 aiScene::~aiScene (this=<optimized out>, this=<optimized out>)
    at /usr/src/debug/assimp/assimp-5.3.1/code/Common/Version.cpp:176
#12 0x00007fffed772fc7 in Assimp::Importer::~Importer (
    this=<optimized out>, this=<optimized out>)
    at /usr/src/debug/assimp/assimp-5.3.1/code/Common/Importer.cpp:191
#13 0x00007ffff7eae9ad in vtkF3DAssimpImporter::vtkInternals::~vtkInternals (this=0x555556143020)
    at /home/michael/dev/f3d/plugins/assimp/module/vtkF3DAssimpImporter.cxx:43
#14 0x00007ffff7eae85b in std::default_delete<vtkF3DAssimpImporter::vtkInternals>::operator() (this=0x555556146b38, __ptr=0x555556143020)
    at /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/13.2.1/../../../../include/c++/13.2.1/bits/unique_ptr.h:99
#15 0x00007ffff7e91b23 in std::unique_ptr<vtkF3DAssimpImporter::vtkInternals, std::default_delete<vtkF3DAssimpImporter::vtkInternals> >::~unique_ptr (this=0x555556146b38)
    at /usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/13.2.1/../../../../include/c++/13.2.1/bits/unique_ptr.h:404
#16 0x00007ffff7e9005e in vtkF3DAssimpImporter::~vtkF3DAssimpImporter (
    this=0x555556146ab0)
    at /home/michael/dev/f3d/plugins/assimp/module/vtkF3DAssimpImporter.cxx:894
```

Reply:
Ok, I understand now. Well  lets link with assimp issue and hope for the best.